doc/nrf/ug_thread.rst

```diff
@@ -0,0 +1,183 @@
+This includes:
+
+* All Thread networking layers:
+
+    * IPv6
+    * 6LoWPAN
+    * IEEE 802.15.4 with MAC security
+    * Mesh Link Establishment
+    * Mesh Routing
+    
+* All device roles
+* Border Router support
+
+
+Required modules
+****************
+
+Thread requires the following Zephyr's modules to properly operate in |NCS|:
+
+* :ref:`zephyr:ieee802154` radio driver - This library is automatically enabled when working with OpenThread on Nordic Semiconductor's Development Kits.
+* :ref:`zephyr:settings` subsystem - This is required to allow Thread to store settings in the non-volatile memory.
+
+Configuration
+*************
+
+To use the Thread protocol in |NCS|, set the following Kconfig options:
+
+* ``CONFIG_NET_L2_OPENTHREAD`` - This option enables the OpenThread stack required for the correct operation of the Thread protocol and allows you to use them.
+* ``CONFIG_CPLUSPLUS`` - This option must be enabled, because OpenThread is implemented in C++.
+* ``CONFIG_REBOOT`` - This option is needed to ensure safe reboot.
+* ``CONFIG_ENTROPY_GENERATOR`` - Required by both OpenThread and radio driver.
+* Options related to the Settings subsystem in the ``storage_partition`` partition of the internal flash:
+
+    * ``CONFIG_SETTINGS``
+    * ``CONFIG_FLASH``
+    * ``CONFIG_FLASH_PAGE_LAYOUT``
+    * ``CONFIG_FLASH_MAP``
+    * ``CONFIG_MPU_ALLOW_FLASH_WRITE``
+    * ``CONFIG_NVS``
+
+* General setting options related to network configuration:
+
+    * ``CONFIG_NETWORKING`` 
+    * ``CONFIG_NET_UDP``
+    * ``CONFIG_NET_SOCKETS``
+
+IPv6 mandatory configuration
+============================
+
+The Thread protocol can only be used with IPv6.
+IPv4 is not supported.
+
+Enable the following options to make Thread work over IPv6:
+
+* ``CONFIG_NET_IPV6``
+* ``CONFIG_NET_CONFIG_NEED_IPV6``
+
+Additionally, since Thread by default registers a considerable amount of IP addresses, the default IPv6 address count values must be increased.
+Set the following options to the provided values:
+
+* ``CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=6``
+* ``CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=8``
+
+Optional configuration options
+==============================
+
+Depending on your configuration needs, you can also set the following options:
+
+* ``CONFIG_NET_SHELL`` - This option enables Zephyr's :ref:`zephyr:shell_label` if you need to access OpenThread CLI.
+* ``CONFIG_COAP`` - This option enables Zephyr's :ref:`zephyr:coap_sock_interface` support.
+* ``CONFIG_OPENTHREAD_COAP`` - This option enables OpenThread's native CoAP API.
+
+You can also change the default values for the following options:
+
+* ``CONFIG_OPENTHREAD_CHANNEL`` - By default set to ``11``.
+  You can set any value ranging from ``11`` to ``26``.
+* ``CONFIG_OPENTHREAD_PANID`` - By default set to ``43981``.
+  You can set any value ranging from ``0`` to ``65535``.
+
+Thread commissioner
```

Comment:
How about calling this paragraph `Thread commissioning` and describing both, commissioner and joiner?
An example of configuration for joiner can be found here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/net/sockets/echo_client/overlay-ot.conf#L49